### ISSUE TITLE
issue in compiling UDP with arrayed type

### ISSUE THREAD

**Issue by aveerubhotla-ventana** [SystemRDL/systemrdl-compiler#147]
following is the sample UDP which was compiling fine on v.1.23.0, but not on v.1.25.2

property myudp {
  type = string[];
  component = reg | field;
};

addrmap top {
  reg {
    myudp = '{"hello"};
    field { } f;
  } regA;
};

error message:

    if issubclass(self.valid_type, rdltypes.references.RefType):
TypeError: issubclass() arg 1 must be a class

**Comment by mpriestleyidex** [comment]
I'm also seeing this. `issubclass()` is being passed an `systemrdl.rdltypes.array.ArrayPlaceholder object`, which (as the error message says) is an instance, not a class.

Error is coming from:
```
  File ".../systemrdl/properties/user_defined.py", line 42, in valid_types
    if issubclass(self.valid_type, rdltypes.references.RefType):
```

I don't know whether the check is wrong, or whether the thing passed to valid_type is wrong to start with.

**Comment by amykyta3** [comment]
Thanks for reporting!
I refactored how UDPs are handled a few versions ago and must have missed something. Will look into it.

**Comment by amykyta3** [comment]
Published fix in v1.25.3
